### PR TITLE
Add xCDAT and cf_xarray to `software.md`

### LINKED ISSUE #361: Add xcdat to the Software page
I have just noticed that the [Software that “Understands” CF Data](http://cfconventions.org/software.html) page does not mention [Xarray Climate Data Analysis Tools  (xCDAT)](https://xcdat.readthedocs.io/)

Can @tomvothecoder suggest what should be added?

### PR DESCRIPTION
**Pull request by tomvothecoder**
- Closes #361
- My markdown formatter also fixed some markdown styling

### CODE REVIEW

File: software.md [hunk 1]
```diff
@@ -114,12 +119,21 @@ Currently supported are:
...
+
+The goal of xCDAT is to provide generalizable features and utilities for simple and robust analysis of climate data. xCDAT's design philosophy is focused on reducing the overhead required to accomplish certain tasks in xarray. xCDAT aims to be compatible with structured grids that are [CF-compliant](https://cfconventions.org/) (e.g., CMIP6). Some key xCDAT features are inspired by or ported from the core CDAT library, while others leverage powerful libraries in the xarray ecosystem (e.g., [xESMF](https://pangeo-xesmf.readthedocs.io/en/latest/), [xgcm](https://xgcm.readthedocs.io/en/latest/index.html), [cf_xarray](https://cf-xarray.readthedocs.io/en/latest/index.html)) to deliver robust APIs.
+
+### `cf_xarray` - Python package for interpreting CF metadata on xarray objects
+
+[cf_xarray](https://cf-xarray.readthedocs.io/en/latest/index.html) mainly provides an accessor (`DataArray.cf` or `Dataset.cf`) that allows you to interpret [Climate and Forecast metadata convention attributes](http://cfconventions.org/) present on [xarray](https://github.com/pydata/xarray) objects.
```
Comment (tomvothecoder) [inline software.md:130]: FYI @dcherian, I'm adding cf_xarray to the list of CF Convention software. Feel free to suggest changes if you want.

Comment (dcherian) [reply software.md:130]: Looks great, thank you!

**Comment by ethanrd** [comment]
Looks good. I will merge this in. Thanks all!

**Comment by tomvothecoder** [comment]
Thanks @ethanrd!

**Pull request merged**